### PR TITLE
feat: support the authorityAttributeIdentifier X.509v3 extension

### PR DESCRIPTION
**Pull request by JonathanWilbur**
This PR adds support for the `authorityAttributeIdentifier` X.509v3 extension as defined in ITU-T Recommendation X.509 for usage in attribute certificates. It's syntax is:

```asn
authorityAttributeIdentifier EXTENSION ::= {
  SYNTAX         AuthorityAttributeIdentifierSyntax
  IDENTIFIED BY  {id-ce-authorityAttributeIdentifier} }

AuthorityAttributeIdentifierSyntax ::= SEQUENCE SIZE (1..MAX) OF AuthAttId

AuthAttId ::= IssuerSerial
```

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by Sashan** [comment]
I think the PR needs rebase.

File: crypto/x509/v3_authattid.c [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ASN1_ITEM_TEMPLATE_END(OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX)
+
+IMPLEMENT_ASN1_FUNCTIONS(OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX)
+
+static int i2r_ISSUER_SERIAL(X509V3_EXT_METHOD *method,
+                             OSSL_ISSUER_SERIAL *iss,
+                             BIO *out, int indent)
+{
+    if (iss->issuer != NULL) {
+        BIO_printf(out, "%*sIssuer Names:\n", indent, "");
```
Comment (Sashan) [inline crypto/x509/v3_authattid.c:30]: Is it intentional not to print `Issuer Names:` header if `->issuer` is NULL. The reason I'm asking we continue to print all other portions (`Issuer Serial`), ...

I think printing either nothing, like function would bail out if `->issuer == NULL` or just empty header would be more friendly to scripts which parse the output. My expectation is the function will either
  - either produce all headers (`Issuer Names:`, `Issuer Serial:` and `Issuer UID:`), and it's fine for me if some values will be missing.
  - or function will produce nothing when `->issuer` is NULL.

Not sure if things I'm asking are reasonable, comply with standard/or existing code.
thank you for considering my suggestion.

**Review by Sashan — CHANGES_REQUESTED** [review]

File: crypto/x509/v3_authattid.c (hunk 1 above)
Comment (JonathanWilbur) [reply crypto/x509/v3_authattid.c:30]: Between the two, I would prefer to display all headers. Would it be bad form for me to print something like `<empty>`?

Comment (Sashan) [reply crypto/x509/v3_authattid.c:30]: From my point of view printing nothing when `->issuer` is NULL is perfectly fine. It's also perhaps easier for script authors to detect if `issuer` attribute is present in the certificate.

Printing `<empty>` (or something similar) is also fine. The only thing that needs to be considered is what is the valid value for `issuer`. For example how one can tell the difference from case when issuer is not present from case when someone deliberately selects issuer name '<empty>' I agree it's example but it just illustrates why printing nothing when there is no issuer attribute is better.

but let's wait for other people what do they think.

Comment (Sashan) [reply crypto/x509/v3_authattid.c:30]: One more thought/reason why I prefer to print nothing when the issuer field is not present: once this change arrives, the output (as I understand) will change for certificates with no issuer too. Not sure if it is desired outcome. Therefore I prefer to print nothing if no issuer is set.

Comment (t8m) [reply crypto/x509/v3_authattid.c:30]: I would be OK with printing `<none>` here. I do not think just immediately returning here is correct.

**Review by t8m — COMMENTED** [review]
Needs a rebase

File: crypto/x509/v3_authattid.c (hunk 1 above)
Comment (JonathanWilbur) [reply crypto/x509/v3_authattid.c:30]: Okay, fixed. I also fixed the same code in `v3_ac_tgt.c`.

**Comment by JonathanWilbur** [comment]
All feedback addressed and ready for review.

**Comment by nhorman** [comment]
I think the style nits can be waived.  Of the 4 issues, 3 are related to the embedded perl code, and the other is a line length issue that doesn't seem to make sense to break up

**Review by nhorman — DISMISSED** [review]

**Comment by JonathanWilbur** [comment]
@nhorman does this mean that the style checks are no longer experimental and I should pay attention to them?

**Comment by nhorman** [comment]
They're not experimental, per-se.  The tool that does the checking (utils/check-format.pl) has been around for some time, but its far from perfect.  The guidance that I would offer is:
1) Yes, pay attention to them
2) If there is a style point that seems out of line (for some definition that you have the agency to determine), make a statement regarding as to why it should be ignored in a comment in the issue.  If there is consensus, the style:waived flag can be set, which will override the check
3) for style nits that do seem relevant, please fix them

We'll use the comments regarding waivers as guidance to try to improve the tool.  The goal of that check was to automatically detect style nits, so that reviews could focus more on the function of the code rather than where spaces and blank lines need to be.

File: util/libcrypto.num [hunk 2]
```diff
@@ -5734,3 +5734,8 @@ EVP_CIPHER_CTX_get_algor                5861	3_4_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_set_algor_params           5862	3_4_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_get_algor_params           5863	3_4_0	EXIST::FUNCTION:
 EVP_PKEY_CTX_get_algor                  5864	3_4_0	EXIST::FUNCTION:
+d2i_OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX  ?	3_4_0	EXIST::FUNCTION:
+i2d_OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX  ?	3_4_0	EXIST::FUNCTION:
+OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX_free ?	3_4_0	EXIST::FUNCTION:
+OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX_new  ?	3_4_0	EXIST::FUNCTION:
+OSSL_AUTHORITY_ATTRIBUTE_ID_SYNTAX_it   ?	3_4_0	EXIST::FUNCTION:
```
Comment (t8m) [inline util/libcrypto.num:5741]: Please change the symbol version numbers here to `3_5_0` as this is not going into 3.4.

Comment (JonathanWilbur) [reply util/libcrypto.num]: Fixed.

**Comment by JonathanWilbur** [comment]
Feedback addressed and ready for review.

**Comment by nhorman** [comment]
approval holds

**Review by t8m — APPROVED** [review]

**Pull request closed without merging**